src/CartesianProduct.jl

```diff
@@ -549,3 +550,36 @@ function vertices_list(cpa::CartesianProductArray{N}
 
     return vlist
 end
+
+
+function linear_map(M::AbstractMatrix{N},
+    cp::CartesianProductArray{N};
+    cond_tol::Number=DEFAULT_COND_TOL,
+    use_inv::Bool=!issparse(M)) where {N<:Real}
+    if !isinvertible(M; cond_tol=cond_tol)
+        error("linear maps for Cartesian product arrays need to be invertible")
+    end
+
+    result = CartesianProductArray(length(cp.array), N)
+    col_st_index, col_end_ind = 0, 0
+    for bi in 1:length(cp.array)
+        col_st_index = col_end_ind + 1
+        col_end_ind += dim(cp.array[bi])
+        n = dim(cp.array[bi])
+        start_ind, end_ind = 1, dim(cp.array[bi])
+        matrices = Vector{Matrix{N}}()
+        while end_ind <= size(M)[1]
+            push!(matrices, M[col_st_index:col_end_ind,start_ind:end_ind])
```

Comment:
This `matrices` vector just creates extra allocations that are not needed. Use a view array for the particular block of `M` where it needs to be applied instead (line 578).